src/gui/options_imp.cpp

```diff
@@ -403,6 +406,18 @@ void options_imp::saveOptions()
     Preferences::setTorrentFileAssoc(checkAssociateTorrents->isChecked());
     Preferences::setMagnetLinkAssoc(checkAssociateMagnetLinks->isChecked());
 #endif
+#ifdef Q_OS_MAC
+    if (checkAssociateTorrents->isChecked()) {
+        Preferences::setTorrentFileAssoc();
+        checkAssociateTorrents->setChecked(Preferences::isTorrentFileAssocSet());
+        checkAssociateTorrents->setEnabled(!checkAssociateTorrents->isChecked());
```

Comment:
@sledgehammer999 When the user pushes Apply or OK button and checkbox is set, association will be applied and checkbox will be greyed out (only after apply action, not after clicking checkbox). If some other program will associate self with torrents defaults our checkboxes will be unset and available to change. Saying simple, it is greyed out when we already have association.